zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/ZiplineLoader.kt

```diff
@@ -147,62 +147,63 @@ class ZiplineLoader internal constructor(
   private val moduleFetchers = listOfNotNull(embeddedFetcher, cachingFetcher ?: httpFetcher)
 
   /**
-   * Loads code from [manifestUrlFlow] each time it emits, skipping loads if the code to load is
-   *    the same as what's already loaded.
+   * Each time when [manifestUrlFlow] emits: Emits a fresh ZiplineManifest and loads Zipline with
+   * the fresh ZiplineManifest.
    *
-   * If the network is unreachable and there is local code newer, then this
-   *   will load local code, either from the embedded directory or the cache.
+   * If the local ZiplineManifest is fresh, emits it and load Zipline without downloading from the
+   * network.
    *
-   * @param manifestUrlFlow a flow that should emit each time a load should be attempted. This
-   *     may emit periodically to trigger polling. It should also emit for loading triggers like
-   *     app launch, app foregrounding, and network connectivity changed.
+   * If the local ZiplineManifest is not fresh, download a new ZiplineManifest from network.
+   *
+   * Newly downloaded ZiplineManifest is always considered fresh.
+   *
+   * If the network is unreachable : returns null, and emits a LoadResult.Failure
+   *
+   * @param manifestUrlFlow a flow that emits whenever a copy of ZiplineManifest is requested by the
+   *  downstream service.
+   * @param freshnessChecker checks if a cached ZiplineManifest is considered fresh.
```

Comment:
first draft, if it looks alright, more test cases will be added 

also a few thoughts I have while working on this: 

1) for local development (of Treehouse), it'll need the continuous loading, one way (which is implied by this change) to support continuous loading is to pass in a `freshnessChecker` that always returns false 

2) related to 1), right now with the current implementation, it's implied that a new download from the network is always considered as fresh and therefore loaded (into Zipline), the question is: is there a danger/potential harm for this assumption? I thought over a few different scenarios, it seems a safe assumption (meaning downstream layers, like Treehouse, would be mitigating with error handling, if the version served by server is not the version intended to be served)